pyfr/plugins/fluidforce.py

```diff
@@ -136,3 +192,31 @@ def __call__(self, intg):
 
             # Flush to disk
             self.outf.flush()
+
+    def stress_tensor(self, u, du):
+        c = self._constants
+
+        # Density and gradient of density
+        rho = u[0]
+        gradrho = du[:, 0]
+
+        # Gradient of momentum
+        gradrhou = du[:, 1:-1]
+
+        # Gradient of velocity
+        gradu = (gradrhou - gradrho[:, None]*u[None, 1:-1]/rho)
+        gradu /= rho
+
+        # Bulk tensor
+        ndims = du.shape[0]
+        bulk = np.eye(ndims)[:, :, None, None]*np.trace(gradu)
+
+        mu = c['mu']
+
+        if self._viscorr == 'sutherland':
+            E = u[-1]
+            cpT = c['gamma']*(E/rho - 0.5*sum(c*c for c in u[1:-1]))
```

Comment:
See if the `sum` can be written in terms of `np.dot` or a relative; might clean things up.